### PR TITLE
Implement removeTransaction function in TxPool - Closes #4861

### LINKED ISSUE #4861: Create remove transactions functionality in the transaction pool
### Description

Transaction pool should remove the transaction in the list by ID

### Motivation

When receiving new block, transactions need to be removed from the pool

### Acceptance Criteria

- It should remove from the corresponding `TransactionList`
- Remove transaction id from `all` transaction map
- Test if transaction list do not contain the deleted transactions
- Test if `all` transaction set does not contain transactions

### Related issues
#4841

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #4861 

### How was it solved?

- [x] It should remove from the corresponding `TransactionList`
- [x] Remove transaction id from `all` transaction map
- [x] Test if transaction list do not contain the deleted transactions
- [x] Test if `all` transaction set does not contain transactions

### How was it tested?

`npm t`

### CODE REVIEW

File: elements/lisk-transaction-pool/src/transaction_pool.ts [hunk 1]
```diff
@@ -174,9 +174,25 @@ export class TransactionPool {
...
+		}
+
+		delete this._allTransactions[tx.id];
+		this._transactionList[
+			getAddressFromPublicKey(foundTx.senderPublicKey)
+		].remove(tx.nonce);
+
+		// Remove from feePriorityQueue
+		this._feePriorityQueue.clear();
+		for (const node of this._feePriorityQueue.getAllNodes()) {
```
Comment (shuse2) [inline elements/lisk-transaction-pool/src/transaction_pool.ts]: I think this is basically same as recreating the new fee priority queue from `this._allTransactions`?
then i think we dont need to add `getAllNodes` and simply recreate?

File: elements/lisk-transaction-pool/src/transaction_pool.ts [hunk 2]
```diff
@@ -186,9 +186,14 @@ export class TransactionPool {
 
 		// Remove from feePriorityQueue
 		this._feePriorityQueue.clear();
-		for (const node of this._feePriorityQueue.getAllNodes()) {
-			if (node.value !== foundTx.id) {
-				this._feePriorityQueue.push(node.key, node.value);
+		for (const txObject of this.getAllTransactions()) {
+			if (txObject.id !== foundTx.id) {
```
Comment (shuse2) [inline elements/lisk-transaction-pool/src/transaction_pool.ts]: since it's removed already above, it will never be `===`

Comment (ishantiw) [reply elements/lisk-transaction-pool/src/transaction_pool.ts]: yes 😅

File: elements/lisk-transaction-pool/src/transaction_pool.ts [hunk 3]
```diff
@@ -174,9 +174,28 @@ export class TransactionPool {
...
+			getAddressFromPublicKey(foundTx.senderPublicKey)
+		].remove(tx.nonce);
+
+		// Remove from feePriorityQueue
+		this._feePriorityQueue.clear();
+		for (const txObject of this.getAllTransactions()) {
+			this._feePriorityQueue.push(
+				txObject.feePriority
+					? txObject.feePriority
+					: this._calculateFeePriority(txObject),
```
Comment (ManuGowda) [inline elements/lisk-transaction-pool/src/transaction_pool.ts]:
```suggestion
				txObject.feePriority ?? this._calculateFeePriority(txObject),
```

File: elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts [hunk 4]
```diff
@@ -237,4 +237,66 @@ describe('TransactionList class', () => {
...
+		it('should return false when a tx id does not exist', async () => {
+			const addStatus = await transactionPool.addTransaction(tx);
+			expect(addStatus).toEqual(true);
+			expect(transactionPool.getAllTransactions().length).toEqual(1);
+			expect(
+				transactionPool['_transactionList'][
+					getAddressFromPublicKey(tx.senderPublicKey)
+				].get(tx.nonce),
+			).toEqual(tx);
+			expect(transactionPool['_feePriorityQueue'].values).toContain(tx.id);
```
Comment (ManuGowda) [inline elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts]: Maybe wrap this in beforeEach hook as you are trying to add trx before removal test

**Pull request merged**